basket/news/tasks.py

```diff
@@ -805,11 +809,43 @@ def process_subhub_event_subscription_charge(data):
         'PMT_Subscription_ID__c': data['subscription_id'],
         'PMT_Transaction_ID__c': data['charge'],
         'RecordTypeId': settings.SUBHUB_OPP_RECORD_TYPE,
-        'Service_Plan__c': data['nickname'],
+        'Service_Plan__c': nickname,
         'StageName': 'Closed Won',
     })
 
 
+@et_task
+def process_subhub_event_subscription_reactivated(data):
+    statsd.incr('news.tasks.process_subhub_event.subscription_reactivated')
+    user_data = get_user_data(payee_id=data['customer_id'],
+                              extra_fields=['id'])
+    if not user_data:
+        statsd.incr('news.tasks.process_subhub_event.subscription_reactivated.user_not_found')
+        raise RetryTask('Could not find user. Try again.')
+
+    nickname = data['nickname']
```

Comment:
Based on my current understanding a missing key error here would result in a retry loop until the task times out. If we get any invalid events with missing keys, what is the expected/desired behavior for this task?

Reply:
No, it would simply throw the exception and be recorded in Sentry and fail. Only specific exception types are retried:

https://github.com/mozmeao/basket/blob/e5fbaa77b9f12b87a5e4b8aecc33c09dd88d01bb/basket/news/tasks.py#L174-L177